module/zfs/vdev_disk.c

```diff
@@ -312,6 +314,9 @@ vdev_disk_open(vdev_t *v, uint64_t *psize, uint64_t *max_psize,
 	v->vdev_tsd = vd;
 	vd->vd_bdev = bdev;
 
+	/* Reset TRIM flag, as underlying device support may have changed */
+	v->vdev_notrim = B_FALSE;
```

Comment:
Good idea, we don't have to guess whether discard is supported by the device.  I'm also not aware of any reason we shouldn't unconditionally do this for open and reopen so I've moved this after the `skip_open` label.  Maybe @skiselkov could shed some additional light on this.